pycroft/lib/net.py

```diff
@@ -24,11 +26,24 @@ def get_free_ip(subnets):
             # Stop argument must be None or an integer: 0 <= x <= sys.maxsize.
             # IPv6 subnets can exceed this boundary on 32 bit python builds.
             min(subnet.address.numhosts - reserved_top - 2, sys.maxsize))
-        unused = (ip for ip in unreserved if ip not in used_ips)
-        try:
-            return next(unused), subnet
-        except StopIteration:
-            continue
+        unused = unused + list(ip for ip in unreserved if ip not in used_ips)
```

Comment:
Eagerly constructing a list of all unused IPs is not optimal, using a generator would be preferable:
```python
for ip in unreserved:
    if ip not in used_ips:
        yield ip
```
OR
```python
for unused_ip in (ip for ip in unreserved if ip not in used_ips):
    yield unused_ip
```